Karl/integrations/nodejs/test.js

```diff
@@ -0,0 +1,176 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Maxim Zhurovich <zhurovich@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+var assert = require('assert');
+var request = require('request');
+var Claire = require('./claire.js');
+
+const testKarlKeepalivesPort = 30001;
+const testKarlFleetViewPort = 30999;
+
+const claireConfig = {
+  service: 'JS.Test',
+  localPort: 30002,
+  cloudInstanceName: 'i-12345',
+  cloudAvailabilityGroup: 'g-1',
+  karlURL: 'http://127.0.0.1:' + String(testKarlKeepalivesPort) + '/',
+  keepaliveInterval: 10000
+}
+
+var customFieldValue = 0;
+
+function statusFiller() {
+  var status = { "custom_status": { custom_field: String(++customFieldValue) } };
+  return status;
+}
+
+describe('Claire JS client test.', function () {
+  var testKarlTerminated = false;
+  var claire;
+
+  it('Check that no service is running on Karl\'s port', function (done) {
+    request({
+      uri: claireConfig.karlURL + '/up'
+    }, function (error, response) {
+      if (error) {
+        done();
+      } else {
+        console.error('Local port ' + String(testKarlKeepalivesPort) + ' is already used.');
+        process.exit(1);
+      }
+    });
+  });
+
+  it('Start test Karl server', function (done) {
+    require('child_process').spawn('.current/karl_server').on('close', function () {
+      testKarlTerminated = true;
+    });
+    var retriesLeft = 10;
+    var checkKarlIsUp = function () {
+      request({
+        uri: claireConfig.karlURL + 'up'
```

Comment:
I suggest to remove the trailing slash from karlURL (the "base URL") and keep in the paths appended to it.